### DISCUSSION TITLE
Creatures spawn inside ice chunks

### DISCUSSION THREAD

**Discussion by A2ur3** [FakeFishGames/Barotrauma#15156] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

I'm using scatter="2000" in cave monsterevent to prevent monster spawning in the same place
but sometimes they will spawn inside ice chunks
SpawnAction can even spawn creatures inside indestructible walls

### Reproduction steps

1. create a cave eventset that spawns many creatures with scatter="2000"
2. they likely spawn inside ice chunk if there exists one

### Bug prevalence

Happens every now and then

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.6.19.1 (Unto the Breach Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by A2ur3** [comment]
![20250221100157](https://github.com/user-attachments/assets/e82ea21b-554f-47ff-abbe-3df675081658)

**Comment by Regalis11** [comment]
Thank you for the report! Does this somehow differ from [Target of MonsterMission can spawn inside level walls](https://github.com/FakeFishGames/Barotrauma/discussions/15654#top)? To me this seems like a duplicate ticket.

**Reply by A2ur3** [reply]
"level walls" refers to indestructible level boundary (rock) and I think monsters have different spawn logic
Monsterevent has `scatter` attribute and it should not choose invalid position (I never saw creatures spawned inside rock at least)
but it doesn't take ice chunks into account?
MonsterMission doesn't seem to support scatter
